specs/AddHostObjectToScriptWithOrigins.md

```diff
@@ -0,0 +1,281 @@
+# Background
+Currently WebView2 supports adding host objects to script from client application
+and then later those objects can be accessed from the JavaScript of the top level frame
+using `window.chrome.webview.hostObjects.{name}`.
+Documentation can be found [here](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2?view=webview2-1.0.705.50#addhostobjecttoscript).
+
+We were asked to support this feature in iframes including
+child frames that don't match the origin of the parent. In this document we describe API
+additions to support host objects in iframes. We'd appreciate your feedback.
+
+
+# Description
+The FrameCreated event is raised when an iframe is created and before starting any
+navigation and content loading. The FrameDeleted event is raised when an iframe is deleted
+or the document containing that iframe is destroyed.
+By providing event handler for frame created event,
+user can obtain the iframe and add host objects to it specifying the list of origins
+using `AddHostObjectToScriptWithOrigins` method.
+When script in the iframe tries to access a host object the origin of the iframe will be
+checked against the provided list. The Frame object has a Name property with the value of the
+name attribute from the iframe html tag declaring it. The NameChanged event is raised when
+corresponding iframe changes its window.name property.
+
+The host object will be removed during iframe deletion automatically.
+
+
+# Examples
+## FrameCreated/FrameDeleted and AddHostObjectToScriptWithOrigins
+
+## .NET, WinRT
+```c#
+void SubscribeToFrameCreated()
```

Comment:
The goal of the sample code is to demonstrate an end developer scenario using these APIs. For the NameChanged and FrameDeleted events below you show hooking up an event handler but there's no indication why or relationship to a scenario.

Samples might be:
* Add host objects to all frames that match particular origin. This would use FrameCreated and AddHostObject..WithOrigins
* Only add host objects to frames with a particular name. Needs to subscribe to name change to decide to call Add or RemoveHostObject depending on what the name's value is. That would use FrameCreated, Name property and NameChagned event, and Add/RemoveHostObjects methods

That's probably enough even though we don't demonstrate FrameDeleted